spec/virtual_total_spec.rb

```diff
@@ -328,4 +362,62 @@ def model_with_children(count)
       end
     end
   end
+
+  describe ".virtual_aggregation" do
+    context "with a standard has_many" do
+      let(:authors) { [author, author2, author3]}
+      let(:author) do
+        Author.create_with_books(1).tap do |author|
+          author.create_books(1, :published => true, :rating => 4)
+          author.create_books(1, :published => true, :rating => 2)
+          author.create_books(1, :published => true)
+        end
+      end
+
+      let(:author2) do
+        Author.create.tap do |author|
+          author.create_books(1, :published => true, :rating => 5)
+        end
+      end
+
+      let(:author3) { Author.create }
+
+      it "calculates sum with one off query" do
+        authors
+        expect do
+          expect(author.sum_recently_published_books_rating).to eq(6)
+          expect(author2.sum_recently_published_books_rating).to eq(5)
+          expect(author3.sum_recently_published_books_rating).to eq(0)
+        end.to match_query_limit_of(3)
+      end
+
+      it "calculates sum from preloaded association" do
+        author.recently_published_books.load
+        author2.recently_published_books.load
+        author3.recently_published_books.load
+
+        expect do
+          expect(author.sum_recently_published_books_rating).to eq(6)
+          expect(author2.sum_recently_published_books_rating).to eq(5)
+          expect(author3.sum_recently_published_books_rating).to be_nil
```

Comment:
Yeah, this is super important that we have this distinction here when comparing to the one above (results in `0` instead).  Will be important when doing refactoring/fixes.  👍